node-websocket-chat/README.md

```diff
@@ -0,0 +1,200 @@
+A client can set up a websocket and listen for events instead of periodically requesting data from the server.
+This design can simplify the client and be more efficient as it avoids the constant creation of new TCP connections.
+
+Without websockets, a web server or API can often be made stateless which has a number of benefits including horizontal scalability.
+The session created by a websocket means that each instance of a web server is responsible for delivering content to its connected clients.
+An event generated in one instance of a webserver may be need to be delivered to clients connected to a different instance.
+Thus, websockets create a data distribution problem for the different instances of a webserver.
+
+
+    +------------+     +------------+
+    | Instance 1 |<--->| Instance 2 |
+    +------------+     +------------+
+           ^                  ^
+           |                  |
+           v                  v
+    +------------+     +------------+
+    |   User A   |     |   User B   |
+    +------------+     +------------+
+
+Take a simple chat application for example.
+Suppose User A is connected to Instance 1 and User B is connected to Instance 2.
+When User A sends a message, User B should receive it and vice versa.
+This means that Instance 1 needs to be aware of all messages received by Instance 2 and vice versa.
+A real chat application would have additional features like multiple rooms/topics, mentions etc.
+However, the basic problem remains:  instances need to send messages to other instances.
+
+Logically, the web server instances and their connected clients are part of a publish/subscribe system.
+There are a number of ways of implementing publish/subscribe and each approach is a tradeoff between cost, efficiency, complexity, and a number of other concerns.
+For example, the instances themselves could just store the events in a database and poll the database for events they need to deliver to their clients.
+Using a database in this way may be sub-optimal so it may be necessary to introduce a broker to store-and-forward events among the instances.
+
+Applications like Redis, RabbitMQ, Apache Kafka, and even MQTT implement the necessary functionality while adding yet another service to be deployed.
+Cloud vendor provide brokers for a fee; for example, Amazon SNS, Google Cloud Pub/Sub, and Azure Service Bus, Event Hub, and Event Grid.
+
+A broker-less approach would have the instances form a messaging routing network, e.g., a minimum spanning tree, amongst themselves.
+This approach favors efficiency and low cost at the expense of complexity.
+Some of the complexity can be mitigated by using an application framework or middleware that was designed for publish/subscribe like OpenDDS.
+The rest of this article shows how to implement a "websocket backend" with OpenDDS using a simple chat application as an example.
```

Comment:
This part introduces OpenDDS -- could use some more background on what it is and how it differs from Kafka et al